lib/sdf-server/src/server/service/graphviz.rs

```diff
@@ -263,46 +389,63 @@ pub async fn nodes_edges(
 
     let root_node_idx = workspace_snapshot.root().await?;
 
-    let nodes = workspace_snapshot
+    let mut nodes = vec![];
+
+    for ( weight, idx ) in workspace_snapshot
         .nodes()
         .await?
-        .into_iter()
-        .map(|(weight, idx)| {
+        .into_iter() {
             node_idx_to_id.insert(idx, weight.id());
             let name = match &weight {
                 NodeWeight::Category(inner) => Some(inner.kind().to_string()),
                 NodeWeight::Func(inner) => Some(inner.name().to_owned()),
                 NodeWeight::Prop(inner) => Some(inner.name().to_owned()),
+                NodeWeight::Component(inner) => Some(Component::get_by_id(&ctx, inner.id().into()).await?.name(&ctx).await?),
+                NodeWeight::AttributePrototypeArgument(inner) => Some(format!(
+                    "APA Targets: {}",
+                    inner
+                        .targets()
+                        .map(|targets| format!(
+                            "\nsource: {}\nto: {}",
+                            targets.source_component_id, targets.destination_component_id
+                        ))
+                        .unwrap_or("".to_string())
+                )),
+                NodeWeight::FuncArgument(inner) => Some(inner.name().to_owned()),
                 _ => None,
```

Comment:
I suspect we will eventually want all of them right? Should those come in this PR or later?

Reply:
This should probably move to the NodeWeight enum in the future too. We discussed it but decided against doing it in this PR